multiproof.go

```diff
@@ -164,13 +164,11 @@ func CheckMultiProof(transcript *common.Transcript, ipaConf *ipa.IPAConfig, proo
 	}
 
 	// Compute E = SUM C_i * (r^i / t - z_i) = SUM C_i * helper_scalars
-	var E banderwagon.Element
-	E.Identity()
-	for i := 0; i < num_queries; i++ {
-		var tmp banderwagon.Element
-		tmp.ScalarMul(Cs[i], &helper_scalars[i])
-		E.Add(&E, &tmp)
+	Csnp := make([]banderwagon.Element, len(Cs))
+	for i := 0; i < len(Cs); i++ {
+		Csnp[i] = *Cs[i]
 	}
+	E := ipa.MultiScalar(Csnp, helper_scalars)
```

Comment:
Note that we have some [TODO optimization deep here](https://github.com/crate-crypto/go-ipa/blob/master/banderwagon/multiexp.go#L17). So we can look into that later after having fresh numbers of the proof benchmarks.

We also have the ever-pending switching to the latest gnark generated code, or using gnark directly as much as possible (e.g: for `Fp` and `Point` probably, and only keep generated `Fr`).  Still in my mind the same blocker of at least having a reference version of go-ipa that looks good enough in all fronts (block execution and proof gen/verif). 

After having a "final" version of go-ipa that looks convincing, we can be a bit more aggressive into things like that to catch any potential regression that isn't acceptable... so we at least have a reference go-ipa that we can always rollback as a reference.

Maybe for acceptable proof gen/verif we need new stuff from gnark, so we might be forced to update sooner. Anyway, we'll see.. just a rant here. I'd like to delete code in this repo. :)

Reply:
> Note that we have some [TODO optimization deep here](https://github.com/crate-crypto/go-ipa/blob/master/banderwagon/multiexp.go#L17). So we can look into that later after having fresh numbers of the proof benchmarks.

I realized now that this TODO shouldn't be impacting this case in particular. The Cs are already in affine form because they were just deserialized from the proofs. So despite they're in projective form, their `Z == 0`.

Reply:
> We also have the ever-pending switching to the latest gnark generated code

OK, you know what... I've already started doing this work again with the latest gnark. 

Soon I'll create a draft PR that we can always rebase on top of master to keep it up to date.
Having that always prepared/updated helps whenever I do some benchmark (for replay or proofs performance), to easily switch there and start comparing.